MiniSimTests/DeviceParserTests.swift

```diff
@@ -105,18 +108,73 @@ class DeviceParserTests: XCTestCase {
     XCTAssertEqual(devices[0].identifier, "mock_adb_id_for_Pixel_3a_API_30_x86")
     XCTAssertTrue(devices[0].booted)
     XCTAssertEqual(devices[0].platform, .android)
+    XCTAssertEqual(devices[0].type, .virtual)
 
     XCTAssertEqual(devices[1].name, "Pixel_4_API_29")
     XCTAssertEqual(devices[1].identifier, "mock_adb_id_for_Pixel_4_API_29")
     XCTAssertTrue(devices[1].booted)
     XCTAssertEqual(devices[1].platform, .android)
+    XCTAssertEqual(devices[1].type, .virtual)
 
     XCTAssertEqual(devices[2].name, "Nexus_5X_API_28")
     XCTAssertEqual(devices[2].identifier, nil)
     XCTAssertFalse(devices[2].booted)
     XCTAssertEqual(devices[2].platform, .android)
+    XCTAssertEqual(devices[2].type, .virtual)
   }
 
+  func testAndroidPhysicalDeviceParser() {
```

Comment:
This is great! We are only missing the iOS physical devices test

Reply:
Do we really need to test ios physical devices? It is a json result and we are not really parsing a raw string with regex but rather structured json. What would you test there?

Reply:
So we are making an assumption that the devicectl command will return this json every time but if the implementation changes or output of the command changes we will catch that in the tests.

I would love to cover MiniSim with more tests and this test will cover all the parsers

Reply:
Alright will try to find some time for it

Reply:
Done